src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -282,8 +282,13 @@ public void clear()
           if (!iterator.hasNext()) {
             int size = tasks.size();
             if (size > 0) {
+              Runnable task;
               do {
-                tasks.pollUnsafe().run();
+                task = tasks.poll();
+                if (task == null) {
+                  break;
+                }
+                task.run();
               }
               while (--size > 0);
```

Comment:
Correct, but the purpose of `size` is to limit the number of tasks executed before calling `select()` and handling network, so it balances between executing tasks and network operations.